---
title: "Class 19: Pertussis Project"
author: "Elena"
format: gfm
editor: visual
toc: TRUE
---

# Investigating Pertussis Cases by Year

```{r, messages=F}
#install.packages("datapasta")
library(ggplot2)
```

To use the package, don't need to call in Library. Go to Addins in the top ribbon.

Highlight and copy data from: https://www.cdc.gov/pertussis/surv-reporting/cases-by-year.html. To paste, go to Addins > Paste as data.frame in R code chunk.

## Q1. With the help of the R “addin” package datapasta assign the CDC pertussis case number data to a data frame called cdc and use ggplot to make a plot of cases numbers over time.

```{r}
 cdc <- data.frame(
                                  Year = c(1922L,1923L,1924L,1925L,
                                           1926L,1927L,1928L,1929L,1930L,
                                           1931L,1932L,1933L,1934L,1935L,
                                           1936L,1937L,1938L,1939L,1940L,1941L,
                                           1942L,1943L,1944L,1945L,1946L,
                                           1947L,1948L,1949L,1950L,1951L,
                                           1952L,1953L,1954L,1955L,1956L,1957L,
                                           1958L,1959L,1960L,1961L,1962L,
                                           1963L,1964L,1965L,1966L,1967L,
                                           1968L,1969L,1970L,1971L,1972L,
                                           1973L,1974L,1975L,1976L,1977L,1978L,
                                           1979L,1980L,1981L,1982L,1983L,
                                           1984L,1985L,1986L,1987L,1988L,
                                           1989L,1990L,1991L,1992L,1993L,1994L,
                                           1995L,1996L,1997L,1998L,1999L,
                                           2000L,2001L,2002L,2003L,2004L,
                                           2005L,2006L,2007L,2008L,2009L,
                                           2010L,2011L,2012L,2013L,2014L,2015L,
                                           2016L,2017L,2018L,2019L),
          Cases = c(107473,164191,165418,
                                           152003,202210,181411,161799,197371,
                                           166914,172559,215343,179135,265269,
                                           180518,147237,214652,227319,
                                           103188,183866,222202,191383,191890,
                                           109873,133792,109860,156517,74715,
                                           69479,120718,68687,45030,37129,
                                           60886,62786,31732,28295,32148,
                                           40005,14809,11468,17749,17135,
                                           13005,6799,7717,9718,4810,3285,
                                           4249,3036,3287,1759,2402,1738,1010,
                                           2177,2063,1623,1730,1248,1895,
                                           2463,2276,3589,4195,2823,3450,
                                           4157,4570,2719,4083,6586,4617,
                                           5137,7796,6564,7405,7298,7867,
                                           7580,9771,11647,25827,25616,15632,
                                           10454,13278,16858,27550,18719,
                                           48277,28639,32971,20762,17972,
                                           18975,15609,18617)
        )
```

```{r}
baseplot <- ggplot(cdc) + 
  aes(x=Year,y=Cases) + 
  geom_point() + geom_line() +
  labs(title="Number of Pertussis Cases in the United States by Year", subtitle="Data Source: CDC")
baseplot
```

# Vaccines

## Q2. Using the ggplot geom_vline() function add lines to your previous plot for the 1946 introduction of the wP vaccine and the 1996 switch to aP vaccine (see example in the hint below). What do you notice?

```{r}
baseplot +
  geom_vline(xintercept=c(1946,1996), lty="dashed", col=c("blue","red")) +
  annotate("text", x=1948, y=240000, label="wP vaccine",angle=90, col="blue") +
  annotate("text", x=1998, y=240000, label="aP vaccine",angle=90, col="red")
```

## Q3. Describe what happened after the introduction of the aP vaccine? Do you have a possible explanation for the observed trend?

After the introduction of the aP vaccine, there is a rise in pertussis cases after a few years. 

Possible explanations include:

- Publication of the now-debunked 1998 Wakefield study that erroneously claimed that vaccines cause autism. This led to an increase in anti-vaccine sentiment

- Vaccine formula not as effective as the original wP vaccine

- Decrease in vaccination and/or public health push for vaccination due to decrease in infections

- Reduction in immunity since the wP vaccine

- Bias and limitation in vaccine trials (study subjects, limited time, limited funding, etc.)

- Increase in bacterial mutations (less likely...there needs to be an evolutionary selection force, which is not immediately clear)


# Exploring CMI-PB Data

```{r}
#install.packages("jsonlite")
library(jsonlite)
subject <- read_json("https://www.cmi-pb.org/api/subject", simplifyVector = TRUE)
head(subject, 3)
```

## Q4. How many aP and wP infancy vaccinated subjects are in the dataset?

47 aP and 49 wP subjects.

```{r}
table(subject$infancy_vac)
```

## Q5. How many Male and Female subjects/patients are in the dataset?

66 female and 30 male.

```{r}
table(subject$biological_sex)
```

## Q6. What is the breakdown of race and biological sex (e.g. number of Asian females, White males etc…)?

- American Indian/Alaska Native: 0F, 1M

- Asian: 18F, 9M

- Black or African American: 2F, 0M

- Native Hawaiian or Other Pacific Islander: 1F, 1M

- White: 27F, 13M

- More Than One Race: 8F, 2M

- Unknown or Not Reported: 10F, 4M

```{r}
table(subject$race, subject$biological_sex)
```

## Q7. Using this approach determine (i) the average age of wP individuals, (ii) the average age of aP individuals; and (iii) are they significantly different?

(i) 36.07532

(ii) 25.23087

(iii) Yes; p-value < 2.2e-16.

```{r, message=F}
library(lubridate)
library(dplyr)
library(tidyverse)
today()
```

Determining Average Age of wP Individuals

```{r}
subject.wP <- subject %>%
  filter(infancy_vac=="wP")
head(subject.wP)
```
```{r}
subject.wP.age <- time_length( today() - ymd(subject.wP$year_of_birth),  "years")
subject.wP.age
```

```{r}
mean(subject.wP.age)
```

Determining Average Age of aP Individuals

```{r}
subject.aP <- subject %>%
  filter(infancy_vac=="aP")
head(subject.aP)
```
```{r}
subject.aP.age <- time_length( today() - ymd(subject.aP$year_of_birth),  "years")
subject.aP.age
```

```{r}
mean(subject.aP.age)
```

```{r}
subject$age <- time_length( today() - ymd(subject$year_of_birth),  "years")
```

Statistical Significance

```{r}
t.test(subject.wP.age,subject.aP.age)
```

## Q8. Determine the age of all individuals at time of boost?

Average 25.60763 years old.

```{r}
age.boost <- time_length( ymd(subject$date_of_boost) - ymd(subject$year_of_birth),  "years")
age.boost
```

```{r}
mean(age.boost)
```

## Q9. With the help of a faceted boxplot or histogram (see below), do you think these two groups are significantly different?

Yes. Visually, those who have received the wP vaccine have a rather large range of ages (late twenties to 50+), whereas those who received the aP vaccine is only 20-30s. However, the data is biased because of the difference in age range.

```{r}
ggplot(subject) +
  aes(age) +
  geom_histogram() +
  facet_wrap(vars(infancy_vac))
```

## Q10. Complete the code to join specimen and subject tables to make a new merged data frame containing all specimen records along with their associated subject details:

```{r}
specimen <- read_json("http://cmi-pb.org/api/specimen", simplifyVector = TRUE) 
titer <- read_json("http://cmi-pb.org/api/ab_titer", simplifyVector = TRUE) 
head(specimen)
head(titer)
```

Joining `subject` and `specimen` tables using the `*_join()` from DPLYR:

```{r}
meta <- inner_join(specimen, subject)
dim(meta)
head(meta)
```

## Q11. Now using the same procedure join meta with titer data so we can further analyze this data in terms of time of visit aP/wP, male/female etc.

```{r}
abdata <- inner_join(titer, meta)
dim(abdata)
```

## Q12. How many specimens (i.e. entries in abdata) do we have for each isotype?

- IgE: 6698

- IgG: 1413

- IgG1: 6141

- IgG2: 6141

- IgG3: 6141

- IgG4: 6141 

```{r}
table(abdata$isotype)
```

## Q13. What do you notice about the number of visit 8 specimens compared to other visits?

Much fewer data entered (80) compared to other visits (4000+).

```{r}
table(abdata$visit)
```

# Examine IgG1 Ab titer levels

```{r}
#Drop visit 8 data as it is incomplete
ig1 <- abdata %>% filter(isotype == "IgG1", visit!=8)
head(ig1)
```

## Q14. Complete the following code to make a summary boxplot of Ab titer levels for all antigens:

```{r}
ggplot(ig1) +
  aes(x=MFI, y=antigen) +
  geom_boxplot() + 
  facet_wrap(vars(visit), nrow=2) +
  labs(title="Antibody Titers", x="Antibody Titer (MFI)", y="Antigen") +
  theme(axis.text.x = element_text(angle = 45, hjust=1))
```

## Q15. What antigens show differences in the level of IgG1 antibody titers recognizing them over time? Why these and not others?

Seems to be an increase in anti-FIM2/3 and anti-FHA antibodies over the number of visits. Both are pertussis bacteria antigens; an increase would indicate that the individuals are responding to the booster vaccine, which include the two antigens. The other antigens are from other pathogens (e.g., TT is tetanus toxin) and therefore the individuals' immune systems should not have a response, unless they were exposed to those pathogens during the course of the study.

```{r}
ggplot(ig1) +
  aes(MFI, antigen, col=infancy_vac ) +
  geom_boxplot() + 
  facet_wrap(vars(visit), nrow=2) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust=1))
```

## Q16. Filter to pull out only two specific antigens for analysis and create a boxplot for each. You can chose any you like. Below I picked a “control” antigen (“Measles”, that is not in our vaccines) and a clear antigen of interest (“FIM2/3”, extra-cellular fimbriae proteins from B. pertussis that participate in substrate attachment).

```{r}
filter(ig1, antigen=="Measles") %>%
  ggplot() +
  aes(MFI, col=infancy_vac) +
  geom_boxplot() +
  facet_wrap(vars(visit)) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust=1))
```

```{r}
filter(ig1, antigen=="FIM2/3") %>%
  ggplot() +
  aes(MFI, col=infancy_vac) +
  geom_boxplot() +
  facet_wrap(vars(visit)) +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust=1))
```

## Q17. What do you notice about these two antigens time courses and the FIM2/3 data in particular?

Anti-FIM2/3 antibodies increase over time, peaking around Visit 5. Anti-measles antibodies are low and do not change over the course of the visits.

## Q18. Do you see any clear difference in aP vs. wP responses?

Antibodies against the aP vaccine seem to have decreased less post-Visit 5 compared to wP. However, the interquartile ranges still overlap at Visit 7 so perhaps a longer timeframe would tell us more about the responses (e.g., if the antibody levels are more sustained over time).

# Obtaining CMI-PB RNASeq data

```{r}
url <- "https://www.cmi-pb.org/api/v2/rnaseq?versioned_ensembl_gene_id=eq.ENSG00000211896.7"
rna <- read_json(url, simplifyVector = TRUE) 
ssrna <- inner_join(rna, meta)
```

## Q19. Make a plot of the time course of gene expression for IGHG1 gene (i.e. a plot of visit vs. tpm)

```{r}
ggplot(ssrna) +
  aes(x=visit, y=tpm, group=subject_id) +
  geom_point() +
  geom_line(alpha=0.2)
```

## Q19. What do you notice about the expression of this gene (i.e. when is it at it’s maximum level)?

Peak expression of this gene is at Visit 4.

## Q20. Does this pattern in time match the trend of antibody titer data? If not, why not?

No; peak of the antibodies is at Visit 5 (need time for gene translation to protein) and seems to be sustained over time.
